### PR TITLE
Canonical factor

### PR DESCRIPTION
**Pull request by yashu-seth**
This PR deals with the CanonicalFactor class.

### CODE REVIEW

File: pgmpy/factors/continuous/CanonicalFactor.py [hunk 1]
```diff
@@ -0,0 +1,480 @@
...
+        variables: list or array-like
+        The variables for wich the distribution is defined.
+
+        K: n x n, 2-d array-like
+
+        h : n x 1, array-like
+
+        g : int, float
+
+        The terms K, h and g are defined parameters for canonical
```
Comment (ankurankan) [inline pgmpy/factors/continuous/CanonicalFactor.py]: @yashu-seth Put this at the top or at the bottom.

File: pgmpy/factors/continuous/CanonicalFactor.py [hunk 2]
```diff
@@ -0,0 +1,480 @@
...
+
+        >>> phi.g
+        -3
+
+        """
+        no_of_var = len(variables)
+
+        if len(h) != no_of_var:
+            raise ValueError("Length of h parameter vector must be equal to the"
+                         "number of variables.")
```
Comment (ankurankan) [inline pgmpy/factors/continuous/CanonicalFactor.py]: @yashu-seth PEP8

**Comment by ankurankan** [comment]
@yashu-seth Yes, changing pdf to an internal attribute makes sense. But changing assignment to pdf won't be consistent with Discrete Factor. So, I would suggest you to leave that. Also you need to finish `to_canonical_factor` in this PR. I am merging this right now. Please create a new PR for these changes.

**Comment by yashu-seth** [comment]
@ankurankan But i still need to add more tests in this PR. What about the
JointGaussianDistribution PR?

**Comment by yashu-seth** [comment]
@ankurankan There are still some things to be done on CanonicalFactor. I found some bugs during adding the tests. I was thinking that the PR #693 would get merged first then I will rebase its changes here.

**Comment by ankurankan** [comment]
@yashu-seth Please add the tests in a new PR. All the changes of that PR were in this PR. Right ?

**Comment by ankurankan** [comment]
@yashu-seth Okay I will merge that now.

**Comment by yashu-seth** [comment]
@ankurankan Yes but I rebased all those commits into 1 commit thinking I will remove that commit when I rebase the changes from PR #693. The work on CanonicalFactor is incomplete. But you can go ahead with PR #693 .

**Comment by yashu-seth** [comment]
@ankurankan Okay.

**Pull request merged**